### PR TITLE
Dev-4788 Add Button Title to CFDA Section

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

Implements a dropdown button that acts as the single cfda sections title when there are multiple cfdas for an award and allows users to switch between cfdas in the single cfda view.

**Technical details:**

N/A

**JIRA Ticket:**
[DEV-4788](https://federal-spending-transparency.atlassian.net/browse/DEV-4788)

**Mockup:**
[Please refer to the ticket.](https://federal-spending-transparency.atlassian.net/browse/DEV-4788)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension) Tested between local and dev with award ASST_NON_CTA01T340_1450 both have 71 issues.
- [N/A] Added Unit Tests for methods in Container Components, reducers, and helper functions `if applicable`
- [x] All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)
- [N/A] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [N/A] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [N/A] Design review complete `if applicable`
- [N/A] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

- [x] Warmfix for titles https://github.com/fedspendingtransparency/usaspending-website/pull/1594 merged and downmerged to dev.

### CODE REVIEW

File: src/js/containers/award/financialAssistance/CFDAVizContainer.jsx [hunk 1]
```diff
@@ -64,6 +64,11 @@ export default class CFDAVizContainer extends React.Component {
         this.setState({ previousView: 'tree', view: 'single', cfda });
     }
 
+    onDropdownClick = (title) => {
```
Comment (MaxwellKendall) [inline src/js/containers/award/financialAssistance/CFDAVizContainer.jsx:67]: Would be nice to have a test for this.

File: src/js/components/award/financialAssistance/SingleCFDA.jsx [hunk 2]
```diff
@@ -19,17 +19,15 @@ const propTypes = {
     })
 };
 
-const SingleCFDA = ({ data }) => {
+const SingleCFDA = ({ currentCfda }) => {
```
Comment (MaxwellKendall) [inline src/js/components/award/financialAssistance/SingleCFDA.jsx:22]: nicely done sir.

File: src/js/components/award/financialAssistance/SingleCFDA.jsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import AwardSection from '../shared/AwardSection';
 import ExpandableAwardSection from '../shared/ExpandableAwardSection';
 
 const propTypes = {
-    data: PropTypes.shape({
+    currentCfda: PropTypes.shape({
```
Comment (MaxwellKendall) [inline src/js/components/award/financialAssistance/SingleCFDA.jsx:8]: much better name

File: src/js/components/award/financialAssistance/CFDAViz.jsx [hunk 4]
```diff
@@ -97,7 +100,34 @@ export default class CFDAViz extends React.Component {
...
+                    .map((x) => ({
+                        name: `${x.cfdaNumber} ${x.cfdaTitle}`,
+                        onClick: onDropdownClick
+                    }));
+                return (<Picker
+                    options={options}
+                    dropdownDirection="right"
+                    backgroundColor="#215493"
+                    selectedOption={`${cfda.cfdaNumber} ${cfda.cfdaTitle}`}
+                    sortFn={() => 1} />);
```
Comment (MaxwellKendall) [inline src/js/components/award/financialAssistance/CFDAViz.jsx:127]: why?

Comment (jonhill13) [reply src/js/components/award/financialAssistance/CFDAViz.jsx:127]: That is a great question it should be this
```
(a, b) => parseFloat(a.cfdaNumber) - parseFloat(b.cfdaNumber);
```
I think we should have a discussion with design on how it should look.

File: src/js/components/award/financialAssistance/SingleCFDA.jsx (hunk 3 above)
Comment (jonhill13) [reply src/js/components/award/financialAssistance/SingleCFDA.jsx:8]: :ballin:

File: src/js/containers/award/financialAssistance/CFDAVizContainer.jsx (hunk 1 above)
Comment (jonhill13) [reply src/js/containers/award/financialAssistance/CFDAVizContainer.jsx:67]: Yeah, this ultimately should change and the Picker should allow for value but until https://github.com/fedspendingtransparency/data-transparency-ui/pull/29 gets merged...

**Pull request merged**